mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -3714,7 +3874,6 @@ pub async fn lp_ordermatch_loop(ctx: MmArc) {
         .expect("CryptoCtx not available")
         .mm2_internal_pubkey_hex();
 
-    let maker_order_timeout = ctx.conf["maker_order_timeout"].as_u64().unwrap_or(MAKER_ORDER_TIMEOUT);
```

Comment:
Why is `maker_order_timeout` not used now, to avoid short-lived orders?
I can see this param is used in tests, maybe could be enabled with "for-tests" feature?

Reply:
Thanks for finding this, there is a test failing due to this removal. Will feature gate this to one of the test features / cfg for sure.

Reply:
Decided to extend the wait in the failing test to 16 seconds instead here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2605/commits/4487d418847af6d11a8eebe446ae61eace3290cf since the timeout in tests is only 15 seconds.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/e8372dc76d50110d1a90107b83618e6965f1ba21/mm2src/mm2_main/src/lp_ordermatch.rs#L139-L140
https://github.com/KomodoPlatform/komodo-defi-framework/blob/e8372dc76d50110d1a90107b83618e6965f1ba21/mm2src/mm2_main/src/lp_ordermatch.rs#L142

Reply:
Test still failing. It might be a different problem other than timeout config in tests, will investigate it now.

Reply:
Ok, `mm2_tests_main` doesn't obey `#[cfg(test)]` while `for-tests` feature requires changes in the CI when running `mm2_tests_main`.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2605/commits/84468c215b7703f46982773d0178ba7de512bbc4 should allow `for-tests` to be used inside `mm2_main` crate while wiring it in the right way. Any integration / docker test now requires `for-tests` feature to be passed, if not passed a compilation error will show. I also added an explicit `[[test]]` targets which allowed running any test from the IDE using the “Run” (play) button works on `mm2_tests_main` or `docker_tests_main`.
The failing `test_order_should_not_be_displayed_when_node_is_down` should now work by using the test value of `MIN_ORDER_KEEP_ALIVE_INTERVAL` as `for-tests` feature was added to it.